jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageReference.java

```diff
@@ -201,6 +201,16 @@ public boolean usesDefaultTag() {
     return DEFAULT_TAG.equals(tag);
   }
 
+  /**
+   * Gets an {@link ImageReference} with the same registry and repository, but a different tag.
+   *
+   * @param tag the new tag
+   * @return an {@link ImageReference} with the same registry/repository and the new tag
+   */
+  public ImageReference retag(String tag) {
```

Comment:
My only hesitation is that this sounds like an action: retag this image as _tag_. `withTag()`?